AGW: SGW: Create separate tunnel for each eNodeB

pshelar opened the pull request:
## Summary

This patch create separate tunnel to each eNodeB on UE attach.
These tunnel are useful to track stats for each eNodeB.

This is feature is flag protected.

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

<!--
    Tag your PR title with the components that it touches.
    E.g. "[lte][agw] Changeset" or "[orc8r][docker] ..."
-->


<!-- Enumerate changes you made and why you made them -->

## Test Plan
- [x]  make test
- [x] integ tests
<!--
    How did you test your change? How do you know it works?
    Add supporting screenshots, terminal pastes, etc. as necessary
-->

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

ardzoht reviewed (approved):
lgtm, please take a look at the merge conflict

pshelar committed:
AGW: SGW: Create separate tunnel for each eNodeB

This create separate tunnel to each eNodeB on UE attach.
This is feature is flag protected.
This can be used for collecting stats for each eNodeB.

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

koolzz commented on lte/gateway/c/oai/tasks/gtpv1-u/gtp_tunnel_openflow.c:
```diff
@@ -30,6 +31,196 @@
+static struct gtp_portno_record gtp_portno_rec;
+
+/**
+ * generate GTP port name from eNodeB IP address
```
[nit] Generate

koolzz reviewed (approved):
LGTM

ulaskozat commented on lte/gateway/c/oai/tasks/gtpv1-u/gtp_tunnel_openflow.c:
```diff
@@ -30,6 +31,196 @@
+      int len = strlen(port_name);
+      // ovsdb dump has port number after portname separated by whitespaces.
+      port_no = atoi(&p[len + 1]);
+      goto out;
```
why not simply use break?

ulaskozat reviewed (approved):
You can avoid the goto statement and section. Please check the test cases that are not added to the sanity yet to make sure that they are not broken.

ulaskozat reviewed (changes requested):
There are test cases that are not part of sanity and uses GTP_PORT. Via grep on GTP_PORT, I can see the following tests dependent on this parameter.

```
s1aptests/test_attach_service_with_multi_pdns_and_bearers_failure.py:    GTP_PORT = 32768
s1aptests/test_attach_detach_rar_tcp_data.py:    GTP_PORT = 32768
s1aptests/test_attach_detach_rar_tcp_data.py:                        "match": {"in_port": self.GTP_PORT},
s1aptests/test_attach_service_with_multi_pdns_and_bearers_multi_ue.py:    GTP_PORT = 32768
s1aptests/test_attach_service_with_multi_pdns_and_bearers_mt_data.py:    GTP_PORT = 32768
s1aptests/test_attach_detach_with_ovs.py:    GTP_PORT = 32768
s1aptests/test_attach_detach_with_ovs.py:                                      "match": {"in_port": self.GTP_PORT}})
s1aptests/s1ap_utils.py:    GTP_PORT = 32768
s1aptests/s1ap_utils.py:                    "match": {"in_port": self.GTP_PORT},
s1aptests/test_attach_asr.py:    GTP_PORT = 32768
s1aptests/test_attach_detach_multiple_rar_tcp_data.py:    GTP_PORT = 32768
s1aptests/test_attach_detach_multiple_rar_tcp_data.py:                        "match": {"in_port": self.GTP_PORT},
s1aptests/test_dedicated_bearer_activation_idle_mode.py:    GTP_PORT = 32768

```

pshelar commented:
I am not able to find test_attach_service_with_multi_pdns_and_bearers_mt_data.py, test_attach_service_with_multi_pdns_and_bearers_multi_ue.py and other tests are already fixed in PR, can you clarify here.

ulaskozat reviewed (approved):
LGTM
